src/apps/intermediate-list/stackable-fees/total-payment-pay.tsx

```diff
@@ -67,15 +68,19 @@ const TotalPaymentPay: FC<TotalPaymentPayProps> = ({
             </>
           }
         />
-        <button
-          type="button"
-          className={paymentButtonClass}
-          onClick={showPaymentButton ? openIntermediatePaymentModal : undefined}
-          disabled={!showPaymentButton}
-        >
-          {t("payText")}
-        </button>
-        {!prePaymentTypeChange && check && (
+        {(postPaymentChangeNotChecked || prePaymentTypeChange) && (
+          <button
+            type="button"
+            className={paymentButtonClass}
+            onClick={
+              showPaymentButton ? openIntermediatePaymentModal : undefined
+            }
+            disabled={!showPaymentButton}
+          >
+            {t("payText")}
+          </button>
+        )}
+        {postPaymentChangeChecked && (
           <button
             type="button"
             className="btn-primary btn-filled btn-small arrow__hover--right-small mt-16"
```

Comment:
I know that this is not related to this change, which is approved anyway.
But I noticed that you use same "base" classes in the two buttons.
So above line 31 you could create a variable like this:
```typescript
const paymentButtonBaseClass = clsx(
    ["btn-primary", "btn-small", "arrow__hover--right-small", "mt-16"],
);
const paymentButtonClass = clsx(
    paymentButtonBaseClass,
    { "btn-outline": !showPaymentButton, "btn-filled": showPaymentButton }
  );
```
and then use the`paymentButtonBaseClass` again in:
```typescript
{postPaymentChangeChecked && (
          <button
            type="button"
            className={paymentButtonBaseClass}